packages/skills/weather_station/handler.py

```diff
@@ -0,0 +1,138 @@
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+#
+# ------------------------------------------------------------------------------
+
+"""This package contains a scaffold of a handler."""
+
+import json
+import time
+from typing import Optional
+
+from aea.configurations.base import ProtocolId
+from aea.mail.base import Envelope
+from aea.protocols.default.message import DefaultMessage
+from aea.protocols.default.serialization import DefaultSerializer
+from aea.protocols.fipa.message import FIPAMessage
+from aea.protocols.fipa.serialization import FIPASerializer
+from aea.protocols.oef.models import Description
+from aea.skills.base import Handler
+from .db_communication import Db_communication
+
+
+class MyWeatherHandler(Handler):
+    """This class scaffolds a handler."""
+
+    SUPPORTED_PROTOCOL = 'fipa'  # type: Optional[ProtocolId]
+
+    def __init__(self, **kwargs):
+        """Initialise the behaviour."""
+        super().__init__(**kwargs)
+        self.fet_price = 0.002
+        self.db = Db_communication("fake")
+        self.fetched_data = []
+
+    def handle_envelope(self, envelope: Envelope) -> None:
+        """
+        Implement the reaction to an envelope.
+
+        :param envelope: the envelope
+        :return: None
+        """
+        msg = FIPASerializer().decode(envelope.message)
+        msg_performative = FIPAMessage.Performative(msg.get('performative'))
+
+        if msg_performative == FIPAMessage.Performative.CFP:
+            self.handle_CFP(msg, envelope.sender)
+        if msg_performative == FIPAMessage.Performative.ACCEPT:
+            self.handle_ACCEPT(envelope.sender)
+        if msg_performative == FIPAMessage.Performative.PROPOSE:
+            pass
+        if msg_performative == FIPAMessage.Performative.DECLINE:
+            print("He refused!!! Make an other offer?")
+            pass
+        if msg_performative == FIPAMessage.Performative.MATCH_ACCEPT:
+            pass
+
+    def teardown(self) -> None:
+        """
+        Implement the handler teardown.
+
+        :return: None
+        """
+        pass
+
+    def handle_CFP(self, msg, sender):
+        """Handle the CFP calls."""
+        fetched_data = self.db.specific_dates("3/10/2019", "4/10/2019")
+
+        if len(fetched_data) >= 1:
+            self.fetched_data = fetched_data
+            totalPrice = self.fet_price * len(fetched_data)
+            proposal = [Description({"Rows": len(fetched_data),
+                                     "Price": totalPrice})]
+            print("[{}]: Sending propose at price: {}".format(
+                sender, totalPrice))
+            proposal_msg = FIPAMessage(message_id=1, dialogue_id=1,
```

Comment:
We should provide constants as CONSTANTS and specify them on the top of the file.